### PR TITLE
chore: add logging for e2e gpu test

### PR DESCRIPTION
**Pull request by eecsliu**
## Description
e2e gpu tests were failing because the created agent was not connecting to master and tests were timing out. The agent failed to start because it was mistakenly looking for a certificate when there isn't one. Confusingly, the certificate name is an empty string. This PR adds some additional logging and a precautionary string trimming function.

## Test Plan
e2e tests should be fixed now

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: agent/internal/agent.go [hunk 1]
```diff
@@ -286,11 +286,13 @@ func (a *agent) tlsConfig() (*tls.Config, error) {
 	}
 
 	var certs []tls.Certificate
-	switch cert, err := a.Options.Security.TLS.ReadClientCertificate(); {
-	case err != nil:
-		return nil, errors.Wrap(err, "failed to read agent certificate file")
-	case cert != nil:
-		certs = append(certs, *cert)
+	if agentCert := a.Options.Security.TLS.ClientCert; agentCert != "" {
```
Comment (ioga) [inline agent/internal/agent.go:289]: I am somewhat groggy at the moment and a bit confused how it works. 
`ReadClientCertificate` itself already checks if `ClientCert == ""`: https://github.com/determined-ai/determined/blob/master/agent/internal/options.go#L112
and then it's supposed to `return nil, nil`, meaning this switch wouldn't hit any cases.
how does the same (?) check on top changes the code flow here?

Comment (eecsliu) [reply agent/internal/agent.go:289]: yeah I'm still looking into this, but the problem was the error "failed to read agent certificate file" and after the change it's not there...

Comment (eecsliu) [reply agent/internal/agent.go]: So I'm just more confused by the error and logically it makes no sense, so I think the best path right now is add logging and a precautionary string trimming step.

File: agent/internal/agent.go [hunk 2]
```diff
@@ -277,7 +277,8 @@ func (a *agent) tlsConfig() (*tls.Config, error) {
 	if certFile := a.Options.Security.TLS.MasterCert; certFile != "" {
 		certData, err := ioutil.ReadFile(certFile) //nolint:gosec
 		if err != nil {
-			return nil, errors.Wrap(err, "failed to read certificate file")
+			msg := fmt.Sprintf("failed to read certificate file %q", certFile)
+			return nil, errors.Wrap(err, msg)
```
Comment (dzhu) [inline agent/internal/agent.go:281]: `errors.Wrapf`

File: agent/internal/agent.go [hunk 3]
```diff
@@ -288,7 +289,9 @@ func (a *agent) tlsConfig() (*tls.Config, error) {
 	var certs []tls.Certificate
 	switch cert, err := a.Options.Security.TLS.ReadClientCertificate(); {
 	case err != nil:
-		return nil, errors.Wrap(err, "failed to read agent certificate file")
+		msg := fmt.Sprintf("failed to read agent certificate file %q or certificate key %q",
+			a.Options.Security.TLS.ClientCert, a.Options.Security.TLS.ClientKey)
+		return nil, errors.Wrap(err, msg)
```
Comment (dzhu) [inline agent/internal/agent.go:294]: `errors.Wrapf`

**Review by dzhu — APPROVED** [review]

**Pull request merged**